glib/src/value.rs

```diff
@@ -1162,6 +1222,51 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_path() {
+        use std::path::{Path, PathBuf};
+
+        let path = Path::new("apple");
+        let pathbuf = path.to_owned();
+        let path_value = path.to_value();
+        let pathbuf_value = pathbuf.to_value();
+
+        assert_eq!(pathbuf_value.get::<PathBuf>(), Ok(Some(pathbuf.clone())));
+        assert_eq!(path_value.get::<PathBuf>(), Ok(Some(pathbuf.clone())));
+        // Drop the backing storage, and compare with str to ensure valid memory management.
+        drop(pathbuf);
```

Comment:
Even though it's dropped, the memory can remain accessible (depending on the page allocation, etc). Not sure if this part of the test is that useful... Or maybe you could simply update the path value?

Reply:
`valgrind` would complain, and at least I'm regularly running our tests with `valgrind`

Reply:
`valgrind` doesn't work on mac and it's really annoying to make it run with `cargo test`. Updating the value and checking that it was updated seems simpler.

Reply:
Yes